visualization/app/codeCharta/ui/nodeContextMenu/nodeContextMenu.component.spec.ts

```diff
@@ -165,6 +165,34 @@ describe("nodeContextMenuController", () => {
 			expect(document.getElementById).toHaveBeenCalledWith("codeMap")
 			expect(elementMock.addEventListener).toHaveBeenCalledWith("wheel", expect.anything(), expect.anything())
 		})
+
+		it("should not shorten the path if it has no sub paths", () => {
+			document.body.addEventListener = jest.fn()
+
+			const elementMock = { addEventListener: jest.fn() }
+			// @ts-ignore
+			jest.spyOn(document, "getElementById").mockImplementation(() => elementMock)
```

Comment:
move these to beforeEach and remove it from each test

Reply:
expect(elementMock.addEventListener).toHaveBeenCalledWith("wheel", expect.anything(), expect.anything()) 
complains that element mock is not defined if i move it to befor each

Reply:
Just move it to the top then:

```ts
describe("onShowNodeContextMenu", () => {
  let elementMock
  beforeEach(() => {
    nodeContextMenuController.setPosition = jest.fn()
    nodeContextMenuController.calculatePosition = jest.fn().mockReturnValue({ x: 1, y: 2 })

    document.body.addEventListener = jest.fn()
    elementMock = { addEventListener: jest.fn() }
    // @ts-ignore
    jest.spyOn(document, "getElementById").mockImplementation(() => elementMock)
  })
  it("should set the correct building after some timeout", () => {
    nodeContextMenuController.onShowNodeContextMenu("/root", NodeType.FOLDER, 42, 24)
    
    expect(nodeContextMenuController["_viewModel"].codeMapNode).toEqual(TEST_DELTA_MAP_A.map)
    expect(nodeContextMenuController["_viewModel"].showNodeContextMenu).toBe(true)
    expect(nodeContextMenuController.calculatePosition).toHaveBeenCalledWith(42, 24)
    expect(nodeContextMenuController.setPosition).toHaveBeenCalledTimes(1)
    expect(nodeContextMenuController.setPosition).toBeCalledWith(1, 2)
    expect(document.body.addEventListener).toHaveBeenNthCalledWith(1, "click", expect.anything(), expect.anything())
    expect(document.body.addEventListener).toHaveBeenNthCalledWith(2, "mousedown", expect.anything(), expect.anything())
    expect(document.getElementById).toHaveBeenCalledWith("codeMap")
    expect(elementMock.addEventListener).toHaveBeenCalledWith("wheel", expect.anything(), expect.anything())
  })
})
```